### PR TITLE
Updated for SDL3 hint changes

### PR DESCRIPTION
**Pull request by slouken**

### CODE REVIEW

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -4285,16 +4297,39 @@ SDL_RenderPresent(SDL_Renderer *renderer)
...
 }
 
-DECLSPEC void SDLCALL
-SDL_DestroyRenderer(SDL_Renderer *renderer)
+DECLSPEC SDL_Texture * SDLCALL
+SDL_CreateTexture(SDL_Renderer * renderer, Uint32 format, int access, int w, int h)
+{
+    SDL_Texture *texture = SDL3_CreateTexture(renderer, format, access, w, h);
+    if (texture) {
+        SDL_SetTextureScaleMode(texture, SDL_GetScaleMode());
```
Comment (sezero) [inline src/sdl2_compat.c:4320]: SDL3_SetTextureScaleMode

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -4285,16 +4297,39 @@ SDL_RenderPresent(SDL_Renderer *renderer)
...
+    return texture;
+}
+
+DECLSPEC SDL_Texture * SDLCALL
+SDL_CreateTextureFromSurface(SDL_Renderer * renderer, SDL_Surface * surface)
 {
-    SDL3_DestroyRenderer(renderer);
+    SDL_Texture *texture = SDL3_CreateTextureFromSurface(renderer, surface);
+    if (texture) {
+        SDL_SetTextureScaleMode(texture, SDL_GetScaleMode());
```
Comment (sezero) [inline src/sdl2_compat.c:4330]: SDL3_SetTextureScaleMode

File: src/sdl2_compat.c [hunk 3]
```diff
@@ -7757,6 +7793,36 @@ SDL_hid_enumerate(unsigned short vendor_id, unsigned short product_id)
...
+SDL_CreateThread(SDL_ThreadFunction fn, const char *name, void *data,
+                 pfnSDL_CurrentBeginThread pfnBeginThread,
+                 pfnSDL_CurrentEndThread pfnEndThread)
+{
+    size_t stacksize = 0;
+    const char *hint = SDL3_GetHint("SDL_THREAD_STACK_SIZE");
+    if (hint) {
+        stacksize = (size_t)SDL_strtoul(hint, NULL, 0);
+    }
+    return SDL_CreateThreadWithStackSize(fn, name, stacksize, data, pfnBeginThread, pfnEndThread);
```
Comment (sezero) [inline src/sdl2_compat.c:7808]: SDL3_CreateThreadWithStackSize

File: src/sdl2_compat.c [hunk 4]
```diff
@@ -7757,6 +7793,36 @@ SDL_hid_enumerate(unsigned short vendor_id, unsigned short product_id)
...
+
+DECLSPEC SDL_Thread *SDLCALL
+SDL_CreateThread(SDL_ThreadFunction fn, const char *name, void *data,
+                 pfnSDL_CurrentBeginThread pfnBeginThread,
+                 pfnSDL_CurrentEndThread pfnEndThread)
+{
+    size_t stacksize = 0;
+    const char *hint = SDL3_GetHint("SDL_THREAD_STACK_SIZE");
+    if (hint) {
+        stacksize = (size_t)SDL_strtoul(hint, NULL, 0);
```
Comment (sezero) [inline src/sdl2_compat.c:7806]: SDL3_strtoul

File: src/sdl2_compat.c [hunk 5]
```diff
@@ -7757,6 +7793,36 @@ SDL_hid_enumerate(unsigned short vendor_id, unsigned short product_id)
...
+
+#else
+
+DECLSPEC SDL_Thread *SDLCALL
+SDL_CreateThread(SDL_ThreadFunction fn, const char *name, void *data)
+{
+    size_t stacksize = 0;
+    const char *hint = SDL3_GetHint("SDL_THREAD_STACK_SIZE");
+    if (hint) {
+        stacksize = (size_t)SDL_strtoul(hint, NULL, 0);
```
Comment (sezero) [inline src/sdl2_compat.c:7819]: SDL3_strtoul

File: src/sdl2_compat.c [hunk 6]
```diff
@@ -7757,6 +7793,36 @@ SDL_hid_enumerate(unsigned short vendor_id, unsigned short product_id)
...
+    }
+    return SDL_CreateThreadWithStackSize(fn, name, stacksize, data);
```
Comment (sezero) [inline src/sdl2_compat.c:7821]: SDL3_CreateThreadWithStackSize

**Comment by sezero** [comment]
OK with the SDL_xx() -> SDL3_xx() changes noted above

**Comment by sezero** [comment]
Pulled this in as is and pushed my clean-up as f90cff775997cf420fbdcb708a807eb1ee0b1360

**Comment by slouken** [comment]
Okay, I was going to clean it up tonight, but thanks!

**Pull request merged**